x-pack/plugins/observability_solution/infra/public/plugin.ts

```diff
@@ -408,26 +401,28 @@ export class Plugin implements InfraClientPluginClass {
 }
 
 const getLogsNavigationEntries = ({
-  capabilities,
+  application,
   config,
   routes,
 }: {
-  capabilities: CoreStart['application']['capabilities'];
+  application: CoreStart['application'];
   config: InfraPublicConfig;
   routes: LogsAppRoutes;
 }) => {
   const entries: NavigationEntry[] = [];
 
   if (!config.featureFlags.logsUIEnabled) return entries;
 
-  if (capabilities.discover?.show && capabilities.fleet?.read) {
-    entries.push({
-      label: 'Explorer',
-      app: 'observability-logs-explorer',
-      path: '/',
-      isBetaFeature: true,
-    });
-  }
+  getLogsExplorerAccessibility$(application).subscribe((isAccessible) => {
+    if (isAccessible) {
+      entries.push({
+        label: 'Explorer',
+        app: 'observability-logs-explorer',
+        path: '/',
+        isBetaFeature: true,
+      });
+    }
```

Comment:
```suggestion
    const logsExplorerExistingIndex = entries.findIndex(
      (entry) => entry.app === 'observability-logs-explorer'
    );

    if (isAccessible && logsExplorerExistingIndex === -1) {
      entries.push({
        label: 'Explorer',
        app: 'observability-logs-explorer',
        path: '/',
        isBetaFeature: true,
      });
    } else if (!isAccessible && logsExplorerExistingIndex > -1) {
      entries.splice(logsExplorerExistingIndex, 1);
    }
```

My not be necessary as Kibana is refreshed after an app's accessibility status change. But better be on the safer side.